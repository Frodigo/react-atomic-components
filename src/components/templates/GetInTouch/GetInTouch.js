import React from 'react';

import PageHeader from '../../organisms/PageHeader';
import Tile from '../../molecues/Tile';
import Newsletter from '../../organisms/Newsletter';
import PageFooter from '../../organisms/PageFooter/PageFooter';
import ShareSection from '../../organisms/ShareSection';
import MobileNav from '../../organisms/MobileNav/MobileNav';
import { items as mobileNavItems } from '../../organisms/MobileNav/MobileNav.config';

import classes from './GetInTouch.module.css';

const Home = (props) => {
    const { ...rest } = props;

    const newsletterProps = {
        title: 'Subscribe to the newsletter',
        subtitle: 'I will send you unique content every Friday.',
        submitUrl: 'https://app.getresponse.com/add_subscriber.html',
        token: 'WMdq9',
        thankYouPageUrl: 'https://marcin-kwiatkowski.com/thank-you',
        footer:
            'Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers',
        onSubmit: (data, e) => e.target.submit()
    };

    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home'
        },
        {
            url: '#',
            label: 'About me'
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch',
            isActive: true

        }
    ];

    return (
        <div className={classes.Home} {...rest}>
            <PageHeader navItems={pageHeaderNavItems} />
            <Tile heading="Let's talk" />
            <div className={classes.content}>
                <p className={classes.paragraph}>
                    Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium
                    dui. In malesuada enim in dolor euismod, id commodo mi
                    consectetur. Curabitur at vestibulum nisi. Nullam vehicula
                    nisi velit. Mauris turpis nisl, molestie ut ipsum et,
                    suscipit vehicula odio.
                </p>
                <figure className={classes.figure}>
                    <img
                        src="https://via.placeholder.com/290x125"
                        alt="figure"
                    />
                </figure>
                <ShareSection heading="Follow me" />
            </div>
            <Newsletter {...newsletterProps} />
            <PageFooter footerColumns={[
                {
                    header: 'Column 1',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                },
                {
                    header: 'Column 2',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                },
                {
                    header: 'Column 3',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                },
                {
                    header: 'Column 4',
                    links: [{
                        url: '#',
                        label: 'lorem ipsum'
                    },{
                        url: '#',
                        label: 'dolor sit amet'
                    }]
                }
            ]} footerBottomLinks={[
                {
                    href: '#',
                    label: 'Link 1'
                },
                {
                    href: '#',
                    label: 'Link 2'
                },
                {
                    href: '#',
                    label: 'Link 3'
                }
            ]} copyright="&copy; 2021 Lorem ipsum" />
            <MobileNav items={mobileNavItems} />
        </div>
    );
};

export default Home;
