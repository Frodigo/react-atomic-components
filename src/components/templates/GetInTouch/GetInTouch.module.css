.content {
    padding: var(--spacing-xxl) var(--spacing-md);
}

.paragraph {
    font-size: 18px;
    line-height: 1.4;
}

.figure {
    margin-top: var(--spacing-xl);
}

.figure img {
    width: 100%;
}

@media (min-width: 768px) {
    .content {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
    }
}
