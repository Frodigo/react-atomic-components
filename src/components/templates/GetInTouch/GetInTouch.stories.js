import React from 'react';

import { default as GetInTouchTemplate } from './GetInTouch';

export default {
    title: 'Templates',
    component: GetInTouchTemplate
};

const Template = (args) => <GetInTouchTemplate {...args} />;

export const GetInTouch = Template.bind({});
