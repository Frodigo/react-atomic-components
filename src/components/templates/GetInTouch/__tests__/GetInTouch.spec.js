import React from 'react';
import { render } from '@testing-library/react';

import GetInTouch from '../GetInTouch';

describe('GetInTouch should be rendered', () => {
    test('GetInTouch renders correctly', () => {
        const { container } = render(<GetInTouch />);

        expect(container).toMatchSnapshot();
    });
});
