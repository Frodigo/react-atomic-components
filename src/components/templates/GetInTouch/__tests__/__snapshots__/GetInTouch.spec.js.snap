// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetInTouch should be rendered GetInTouch renders correctly 1`] = `
<div>
  <div
    class="Home"
  >
    <header
      class="pageHeader"
    >
      <div
        class="container"
      >
        <a
          class="link"
          href="/"
        >
          <img />
        </a>
        <nav
          class="primaryNav"
        >
          <a
            class=" navLink"
            href="#"
          >
            Home
          </a>
          <a
            class=" navLink"
            href="#"
          >
            About me
          </a>
          <a
            class=" navLink"
            href="#"
          >
            Blog
          </a>
          <a
            class="navLinkActive navLink"
            href="#"
          >
            Get in touch
          </a>
        </nav>
      </div>
    </header>
    <section
      class="tile"
    >
      <h2
        class="heading"
      >
        Let's talk
      </h2>
    </section>
    <div
      class="content"
    >
      <p
        class="paragraph"
      >
        Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio.
      </p>
      <figure
        class="figure"
      >
        <img
          alt="figure"
          src="https://via.placeholder.com/290x125"
        />
      </figure>
      <div
        class="shareSection"
      >
        <h4
          class=""
        >
          Follow me
        </h4>
      </div>
    </div>
    <form
      action="https://app.getresponse.com/add_subscriber.html"
      class="newsletter"
      id="newsletter"
      method="post"
    >
      <div
        class="newsletterContent"
      >
        <h3
          class="title"
        >
          Subscribe to the newsletter
        </h3>
        <p
          class="paraghaph"
        >
          I will send you unique content every Friday.
        </p>
        <div
          class="fieldset"
        >
          <div
            class="field"
          >
            <input
              class="input null"
              data-testid="input"
              name="first_name"
              placeholder="Your name"
              type="text"
              value=""
            />
          </div>
          <div
            class="field"
          >
            <input
              class="input null"
              data-testid="input"
              name="email"
              placeholder="Your email"
              type="email"
              value=""
            />
          </div>
        </div>
        <input
          name="campaign_token"
          type="hidden"
          value="WMdq9"
        />
        <input
          name="thankyou_url"
          type="hidden"
          value="https://marcin-kwiatkowski.com/thank-you"
        />
        <input
          name="start_day"
          type="hidden"
          value="0"
        />
        <footer
          class="footer"
        >
          <p>
            Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers
          </p>
          <button
            class="primary"
            type="submit"
          >
            Subscribe
          </button>
        </footer>
      </div>
    </form>
    <footer
      class="pageFooter"
    >
      <div
        class="columns"
      >
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 1
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 2
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 3
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 4
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
      </div>
      <div>
        <ul
          class="bottomNav"
        >
          <li>
            <a
              href="#"
            >
              Link 1
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Link 2
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Link 3
            </a>
          </li>
        </ul>
        <p
          class="copyright"
        >
          Â© 2021 Lorem ipsum
        </p>
      </div>
    </footer>
    <nav
      class="mobileNav"
    >
      <a
        href="/"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Home"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Home
          </span>
        </span>
      </a>
      <a
        href="/blog"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Blog"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Blog
          </span>
        </span>
      </a>
      <a
        href="/about-me"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="About"
            src="[object Object]"
          />
          <span
            class="label"
          >
            About
          </span>
        </span>
      </a>
      <a
        href="/get-in-touch"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Contact"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Contact
          </span>
        </span>
      </a>
      <a
        href="#newsletter"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Subscribe"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Subscribe
          </span>
        </span>
      </a>
    </nav>
  </div>
</div>
`;
